File: Assets/Scripts/LuaScripts/UILogic/Loading/LoadingCtrl.lua
-- ******************************************************************
--@file         LoadingCtrl.lua
--@brief        加载窗口的controller
--@author       yufulao, yufulao@qq.com
--@createTime   2024/5/18 1:25
-- ******************************************************************
---@brief
---@class LoadingCtrl : UIBaseCtrl
LoadingCtrl = Class("LoadingCtrl", UIBaseCtrl)

---@brief 初始化回调
---@public
function LoadingCtrl:OnInit(...)
end

---@brief 打开窗口
---@public
function LoadingCtrl:OpenRoot(...)
    self.view:OpenRoot(...)
end

---@brief 关闭窗口
---@public
function LoadingCtrl:CloseRoot()
    self.view:CloseRoot()
end

---@brief 绑定事件
---@public
function LoadingCtrl:BindEvent()
end


return LoadingCtrl

File: Assets/Scripts/UI/UILogic/LoadingView/LoadingCtrl.cs
// ******************************************************************
//@file         LoadingCtrl.cs
//@brief        加载界面的controller
//@author       yufulao, yufulao@qq.com
//@createTime   2024.05.18 01:35:35
// ******************************************************************

namespace Yu
{
    public class LoadingCtrl : UICtrlBase
    {
        private LoadingModel _model;
        private LoadingView _view;


        public override void OnInit(params object[] param)
        {
            _model = new LoadingModel();
            _view = GetComponent<LoadingView>();
        }

        public override void OpenRoot(params object[] param)
        {
            _view.OpenWindow();
        }

        public override void CloseRoot()
        {
            _view.CloseWindow();
        }

        public override void BindEvent()
        {
        }

    }
}